git/working.go

```diff
@@ -48,55 +49,13 @@ type CommitAction struct {
 	SigningKey string
 }
 
-// TagAction - struct holding tag information
-type TagAction struct {
+// SyncMarkerAction - struct holding tag information
+type SyncMarkerAction struct {
 	Revision   string
 	Message    string
 	SigningKey string
 }
 
-// Clone returns a local working clone of the sync'ed `*Repo`, using
```

Comment:
A couple good reasons, I think:

1. because it's a method on `Repo` and every other method on `Repo` is in the `repo.go` file.
1. because it carries a variable `ErrReadOnly` with it into `working.go` that isn't used elsewhere with the file whereas the rest of the `Repo` errors (`ErrNoChanges`, `ErrNoConfig`, `ErrNotCloned`, and `ErrClonedOnly`) are together in `repo.go`
1. `Repo.Clone` is not called anywhere in `working.go` whereas `repo.go` is the place where the logic that `Repo.Clone` actually uses (i.e. `Repo.createClone`) lives.  Co-locating these two methods therefore seems ideal
1. every other method in that file is on `Checkout`, having a single method for `Repo` here when the rest are in the `repo.go` file in the same package seemed like an oversight.

that said, I am more than happy to move it, but, to your point, the fact that `Repo.Clone` lived in `working.go` alongside nothing other than other methods on `Checkout` seemed like an incidental, if not accidental, placement in the codebase.

---

To be clear, I don't intend to start a discussion over these kinds of things - I really don't deeply care I was just trying to improve the codebase.  I did have a rationale for doing so and now that you've heard what I was thinking if you still disagree you don't need to say anything more than just: "change it"

Reply:
It was here because it's the means of obtaining a `*Checkout`, so something like a constructor. But I'm ambivalent.

Reply:
ok, if you're ambivalent I'll leave this `Repo` method with the other `Repo` methods for the specific reasons mentioned above. thanks :)